spinalcordtoolbox/reports/qc.py

```diff
@@ -660,7 +653,13 @@ def _update_html_assets(self, json_data):
         with io.open(os.path.join(assets_path, 'index.html'), encoding="utf-8") as template_index:
             template = Template(template_index.read())
             output = template.substitute(sct_json_data=json.dumps(json_data))
-            io.open(os.path.join(dest_path, 'index.html'), 'w', encoding="utf-8").write(output)
+
+        # Use locking when writing to `index.html` only (since this is the only file that multiple processes
+        # would simultaneously write to)
+        dest_file = io.open(os.path.join(dest_path, 'index.html'), 'w', encoding="utf-8")
+        portalocker.lock(dest_file, portalocker.LOCK_EX)
+        dest_file.write(output)
+        dest_file.close()
```

Comment:
`portalocker.LOCK_EX` refers to an exclusive lock, see: https://portalocker.readthedocs.io/en/latest/portalocker.constants.html

Reply:
I believe that on Unix-likes, closing a file handle also releases fcntl locks, but I'm not sure about Windows. The [portalocker examples section](https://portalocker.readthedocs.io/en/latest/#examples) seems to say that it's best to
1. `flush()` the file
2. `unlock()` it explicitly
3. `close()` it.